src/wp-admin/includes/ajax-actions.php

```diff
@@ -2994,13 +2994,9 @@ function wp_ajax_query_attachments() {
 
 	$posts = array_map( 'wp_prepare_attachment_for_js', $query->posts );
 	$posts = array_filter( $posts );
+	header( 'X-WP-TotalPages: ' . $query->found_posts );
```

Comment:
@adamsilverstein this is incorrect. This is total results and not total pages. The header `X-WP-Total` should be used for the number of posts. 

I don't see the harm in having `X-WP-TotalPages` and `X-WP-Total`. You can just copy and paste this basically. 

https://github.com/WordPress/wordpress-develop/blob/852a2c216c4c99cd709806b8c700839333ee1f14/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L377-L402

Reply:
Fixed